libtransact/src/scheduler/serial/core.rs

```diff
@@ -156,92 +170,150 @@ impl SchedulerCore {
+            .pop()
+            .expect("There are no transactions left in the queue for the current batch");
+        let transaction_id = transaction.header_signature().into();
+        let transaction_pair = match transaction.into_pair() {
+            Ok(pair) => pair,
+            Err(err) => {
+                self.invalidate_current_batch(InvalidTransactionResult {
+                    transaction_id,
+                    error_message: format!("ill-formed transaction: {}", err),
+                    error_data: vec![],
+                });
+                self.send_batch_result();
+                return Ok(());
             }
+        };
 
-            let transaction = match pair.batch().transactions()[0].clone().into_pair() {
-                Ok(transaction) => transaction,
-                Err(err) => {
-                    // An error can occur here if the header of the
-                    // transaction cannot be deserialized. If this occurs,
-                    // we panic as handling this is not currently implemented.
-                    unimplemented!("cannot handle ill-formed transaction: {}", err)
-                }
-            };
-
-            let context_id = match self.previous_context {
-                Some(previous_context_id) => self
-                    .context_lifecycle
-                    .create_context(&[previous_context_id], &self.state_id),
-                None => self.context_lifecycle.create_context(&[], &self.state_id),
-            };
-
-            self.current_batch = Some(pair);
-            self.execution_tx
-                .send(ExecutionTask::new(transaction, context_id))?
-        }
+        let context_id = match self.previous_context {
+            Some(previous_context_id) => self
+                .context_lifecycle
+                .create_context(&[previous_context_id], &self.state_id),
+            None => self.context_lifecycle.create_context(&[], &self.state_id),
+        };
+
+        self.current_txn = Some(transaction_pair.clone());
+        self.execution_tx
+            .send(ExecutionTask::new(transaction_pair, context_id))?;
+        self.next_ready = false;
 
         Ok(())
     }
 
+    fn invalidate_current_batch(&mut self, invalid_result: InvalidTransactionResult) {
+        // Invalidate all previously executed transactions in the batch
+        self.txn_results = self
+            .txn_results
+            .iter()
+            .map(|result| match result {
+                TransactionExecutionResult::Valid(receipt) => {
+                    TransactionExecutionResult::Invalid(InvalidTransactionResult {
+                        transaction_id: receipt.transaction_id.clone(),
+                        error_message: "batch invalid".into(),
+                        error_data: vec![],
+                    })
+                }
+                TransactionExecutionResult::Invalid(_) => {
+                    panic!("Should not already have an invalid result");
```

Comment:
Add some more comments around this describing why it can't happen.

Reply:
Added